### ISSUE TITLE
Store model in a different way?

### ISSUE THREAD

**Issue by hsonne** [KWB-R/fhpredict#24]
Storing a rstanarm-model in the database now returns an error:
```
> fhpredict::api_add_model(user_id, spot_id, result[[3]], comment = "great!")
Converting model object to text ... ok. (1.27s) 
POST request 'users/3/bathingspots/18/models' returned with error:
- status code: 413
- category: Client error
- reason: Request Entity Too Large
- message: Client error: (413) Request Entity Too Large
```

**Comment by fabianmoronzirfas** [comment]
I can implement in the  upcoming week the binary upload to the S3 bucket

**Comment by hsonne** [comment]
I tested the upload of an image but got the following error. @fabianmoronzirfas do you know what is wrong?
```
> file <- "C:/Users/hsonne/Documents/R/win-library/3.6/recipes/help/figures/logo.png"
> model_data <- httr::upload_file(file)
> fhpredict::postgres_post(path, body = model_data)
POST request 'users/3/bathingspots/18/images/upload' returned with error:
- status code: 500
- category: Server error
- reason: Internal Server Error
- message: Server error: (500) Internal Server Error

list()
attr(,"response")
Response [http://flsshygn-dev.eu-central-1.elasticbeanstalk.com/api/v1/users/3/bathingspots/18/images/upload]
  Date: 2019-09-16 11:42
  Status: 500
  Content-Type: application/json; charset=utf-8
  Size: 327 B
{
  "error": {
    "message": "Cannot read property 'location' of undefined",
    "stack": "TypeError: Cannot read property 'location' of undefined\n    at exports.postFile (/usr/app/dist/lib/request...
  }
```

**Comment by fabianmoronzirfas** [comment]
Several things that need to be taken care of

1. It has to by a POST of type multipart/form-data see [this](https://rdrr.io/cran/httr/man/POST.html#heading-2) and [this](https://cran.r-project.org/web/packages/httr/vignettes/quickstart.html#request-body)
2. The location issue you have seems to originate from [here](https://github.com/technologiestiftung/flusshygiene-postgres-api/blob/master/src/lib/request-handlers/bathingspots/collections/post-file.ts#L97). Which seems to me like the file is not coming in if it us `undefined`
3. The key for the file (path) value needs to be `upload`


In Curl the call would be like this:

```bash
curl --request POST \
  --url http://localhost:5004/api/v1/users/3/bathingspots/18/images/upload \
  --header 'Accept: application/json' \
  --header 'Content-Type: application/json' \
  --header 'cache-control: no-cache' \
  --header 'content-type: multipart/form-data; boundary=----WebKitFormBoundary7MA4YWxkTrZu0gW' \
  --form 'upload=@/Users/icke/Desktop/Screenshot 2019-08-27 at 16.28.29.png'
```

In Python 3 like this

```python
import http.client

conn = http.client.HTTPConnection("localhost")

payload = "------WebKitFormBoundary7MA4YWxkTrZu0gW\r\nContent-Disposition: form-data; name=\"upload\"; filename=\"Screenshot 2019-08-27 at 16.28.29.png\"\r\nContent-Type: image/png\r\n\r\n\r\n------WebKitFormBoundary7MA4YWxkTrZu0gW--"

headers = {
    'content-type': "multipart/form-data; boundary=----WebKitFormBoundary7MA4YWxkTrZu0gW",
    'Accept': "application/json",
    'Content-Type': "application/json",
    'cache-control': "no-cache",
    'Postman-Token': "59041a77-cf98-4b7e-9319-5c8457de15e4"
    }

conn.request("POST", "api,v1,users,3,bathingspots,18,images,upload", payload, headers)

res = conn.getresponse()
data = res.read()

print(data.decode("utf-8"))
```

I think the webkit form boundary stuff could be omitted?

**Comment by fabianmoronzirfas** [comment]
BTW. The following error:

```bash
"Cannot read property 'location' of undefined",
```

Occurs for me when the path to the file I want to upload is not right.

**Comment by hsonne** [comment]
I think that this is solved since ef738c45
